### ISSUE TITLE
Layout/LineContinuationLeadingSpace only supports ' and " as string terminators

### ISSUE THREAD

**Issue by rdubya** [rubocop/rubocop#11356]
Currently strings that do not use ' or " to enclose them cause Rubocop to throw an exception.

--------

## Expected behavior

The cop does not flag it as an offense.

## Actual behavior

It throws an exception:
```
An error occurred while Layout/LineContinuationLeadingSpace cop was inspecting /srv/app/spec/controllers/my_controller_spec.rb:109:23.
undefined method `length' for nil:NilClass
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/layout/line_continuation_leading_space.rb:98:in `block in investigate_trailing_style'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/base.rb:362:in `correct'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/base.rb:126:in `add_offense'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/layout/line_continuation_leading_space.rb:97:in `investigate_trailing_style'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/layout/line_continuation_leading_space.rb:69:in `block in on_dstr'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/layout/line_continuation_leading_space.rb:61:in `each'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/layout/line_continuation_leading_space.rb:61:in `each_cons'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/layout/line_continuation_leading_space.rb:61:in `on_dstr'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/commissioner.rb:101:in `public_send'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/commissioner.rb:101:in `block (2 levels) in trigger_responding_cops'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/commissioner.rb:165:in `with_cop_error_handling'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/commissioner.rb:100:in `block in trigger_responding_cops'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/commissioner.rb:99:in `each'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/commissioner.rb:99:in `trigger_responding_cops'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/commissioner.rb:69:in `on_dstr'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:135:in `on_lvasgn'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/commissioner.rb:71:in `on_lvasgn'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `each'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:158:in `on_block'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `each'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:158:in `on_block'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `each'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:158:in `on_block'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `each'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:158:in `on_block'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/bundle/ruby/3.0.0/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:20:in `walk'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/commissioner.rb:87:in `investigate'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/team.rb:154:in `investigate_partial'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cop/team.rb:82:in `investigate'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/runner.rb:315:in `inspect_file'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/runner.rb:259:in `block in do_inspection_loop'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/runner.rb:293:in `block in iterate_until_no_changes'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/runner.rb:286:in `loop'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/runner.rb:286:in `iterate_until_no_changes'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/runner.rb:255:in `do_inspection_loop'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/runner.rb:138:in `block in file_offenses'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/runner.rb:163:in `file_offense_cache'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/runner.rb:137:in `file_offenses'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/runner.rb:128:in `process_file'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/runner.rb:109:in `block in each_inspected_file'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/runner.rb:108:in `each'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/runner.rb:108:in `reduce'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/runner.rb:108:in `each_inspected_file'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/runner.rb:94:in `inspect_files'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/runner.rb:47:in `run'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cli/command.rb:11:in `run'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cli/environment.rb:18:in `run'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cli/command/auto_generate_config.rb:95:in `execute_runner'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cli/command/auto_generate_config.rb:76:in `run_all_cops'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cli/command/auto_generate_config.rb:24:in `run'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cli/command.rb:11:in `run'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cli/environment.rb:18:in `run'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cli.rb:72:in `run_command'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cli.rb:77:in `execute_runners'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/lib/rubocop/cli.rb:48:in `run'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/exe/rubocop:19:in `block in <top (required)>'
/bundle/ruby/3.0.0/gems/benchmark-0.2.0/lib/benchmark.rb:311:in `realtime'
/bundle/ruby/3.0.0/gems/rubocop-1.41.1/exe/rubocop:19:in `<top (required)>'
/bundle/bin/rubocop:27:in `load'
/bundle/bin/rubocop:27:in `<main>'
```

## Steps to reproduce the problem

A file with something like the following will cause the exception:

```
html = %(<input value="#{my_var}" autocomplete="off") \
  ' type="hidden" name="xyz"' \
  ' id="theid" />'
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.41.1 (using Parser 3.1.3.0, rubocop-ast 1.24.1, running on ruby 3.0.4) [aarch64-linux]
  - rubocop-graphql 0.18.0
  - rubocop-performance 1.15.2
  - rubocop-rails 2.17.4
  - rubocop-rspec 2.16.0
```

**Comment by koic** [comment]
This issue has been resolved in RuboCop 1.63.4. Please upgrade to the latest version of RuboCop. Thank you.
